blockchain/consensus/deleg_cns/src/validation.rs

```diff
@@ -92,10 +92,20 @@ fn validate_miner<DB>(
 where
     DB: BlockStore + Send + Sync + 'static,
 {
+    use DelegatedConsensusError::*;
     let miner_addr = header.miner_address();
+
+    // Workaround for the bug where Forest strips the network type from the Address
```

Comment:
Can we? :thinking: 
In effect, yes, because even if we use `t` it becomes `f` internally, the network part is almost meaningless. I also found that even though I was creating `t01000` address in `genesis.json`, the address in the state tree got resolved into a work address prefixed by `f`. 
I more or less stopped trying to understand whether this is intentional at that point.